CMakeLists.txt

```diff
@@ -73,6 +73,16 @@ if (CMAKE_BUILD_TYPE STREQUAL "")
     set( CMAKE_BUILD_TYPE "RelWithDebInfo" )
 endif()
 
+option(ENABLE_WSTRING "Enable std::wstring support" ON)
+if (NOT ENABLE_WSTRING)
+    add_definitions(-DPOCO_NO_WSTRING)
+endif()
+
+option(ENABLE_FPENVIRONMENT "Enable floating-point support" ON)
+if (NOT ENABLE_FPENVIRONMENT)
+    add_definitions(-DPOCO_NO_FPENVIRONMENT)
```

Comment:
Please use `target_compile_definitions` and add this to the Foundation target. Use also PUBLIC as scope, because this define is used in FPEnvironment.h and should be exported via Foundation_Config.cmake so that "users" has the correct compile definitions to use PocoFoundation.libs ...